### DISCUSSION TITLE
Smoke sensor TS0601 (_TZE200_uebojraa) doesn't report smoke

### DISCUSSION THREAD

**Discussion by Roelski** [Koenkk/zigbee2mqtt#10373] [Devices]
I have 2 smoke sensors (TS0601 = _TZE200_uebojraa).
( https://github.com/Koenkk/zigbee2mqtt.io/blob/master/docs/devices/TS0601_smoke.md )

They are paired fine, but as far as i can test, they never report smoke.

Pressing test button on side gives:
zigbee2mqtt             | 2021-12-24T18:18:14.024254149Z Zigbee2MQTT:debug 2021-12-24 19:18:14: Received Zigbee message from 'Smoke-Woonkamer', type 'commandDataReport', cluster 'manuSpecificTuya', data '{"data":{"data":[0],"type":"Buffer"},"datatype":4,"dp":1,"fn":0,"seq":11520}' from endpoint 1 with groupID null
zigbee2mqtt             | 2021-12-24T18:18:14.035912285Z Zigbee2MQTT:debug 2021-12-24 19:18:14: No converter available for 'TS0601_smoke' with cluster 'manuSpecificTuya' and type 'commandDataReport' and data '{"data":{"data":[0],"type":"Buffer"},"datatype":4,"dp":1,"fn":0,"seq":11520}'
zigbee2mqtt             | 2021-12-24T18:18:14.058850853Z Zigbee2MQTT:info  2021-12-24 19:18:14: MQTT publish: topic 'zigbee2mqtt/Smoke-Woonkamer', payload '{"battery_low":null,"last_seen":"2021-12-24T19:18:14+01:00","linkquality":255,"smoke":null}'
zigbee2mqtt             | 2021-12-24T18:18:15.482588348Z Zigbee2MQTT:debug 2021-12-24 19:18:15: Received Zigbee message from 'Smoke-Woonkamer', type 'commandDataReport', cluster 'manuSpecificTuya', data '{"data":{"data":[1],"type":"Buffer"},"datatype":4,"dp":1,"fn":0,"seq":11776}' from endpoint 1 with groupID null
zigbee2mqtt             | 2021-12-24T18:18:15.493820609Z Zigbee2MQTT:debug 2021-12-24 19:18:15: No converter available for 'TS0601_smoke' with cluster 'manuSpecificTuya' and type 'commandDataReport' and data '{"data":{"data":[1],"type":"Buffer"},"datatype":4,"dp":1,"fn":0,"seq":11776}'
zigbee2mqtt             | 2021-12-24T18:18:15.510105098Z Zigbee2MQTT:info  2021-12-24 19:18:15: MQTT publish: topic 'zigbee2mqtt/Smoke-Woonkamer', payload '{"battery_low":null,"last_seen":"2021-12-24T19:18:15+01:00","linkquality":255,"smoke":null}'

Pressing again for second test
zigbee2mqtt             | 2021-12-24T18:23:32.603980894Z Zigbee2MQTT:debug 2021-12-24 19:23:32: Received Zigbee message from 'Smoke-Woonkamer', type 'commandDataReport', cluster 'manuSpecificTuya', data '{"data":{"data":[0],"type":"Buffer"},"datatype":4,"dp":1,"fn":0,"seq":12032}' from endpoint 1 with groupID null
zigbee2mqtt             | 2021-12-24T18:23:32.619597884Z Zigbee2MQTT:debug 2021-12-24 19:23:32: No converter available for 'TS0601_smoke' with cluster 'manuSpecificTuya' and type 'commandDataReport' and data '{"data":{"data":[0],"type":"Buffer"},"datatype":4,"dp":1,"fn":0,"seq":12032}'
zigbee2mqtt             | 2021-12-24T18:23:32.637235688Z Zigbee2MQTT:info  2021-12-24 19:23:32: MQTT publish: topic 'zigbee2mqtt/Smoke-Woonkamer', payload '{"battery_low":null,"last_seen":"2021-12-24T19:23:32+01:00","linkquality":255,"smoke":null}'
zigbee2mqtt             | 2021-12-24T18:23:34.066938920Z Zigbee2MQTT:debug 2021-12-24 19:23:34: Received Zigbee message from 'Smoke-Woonkamer', type 'commandDataReport', cluster 'manuSpecificTuya', data '{"data":{"data":[1],"type":"Buffer"},"datatype":4,"dp":1,"fn":0,"seq":12288}' from endpoint 1 with groupID null
zigbee2mqtt             | 2021-12-24T18:23:34.077227908Z Zigbee2MQTT:debug 2021-12-24 19:23:34: No converter available for 'TS0601_smoke' with cluster 'manuSpecificTuya' and type 'commandDataReport' and data '{"data":{"data":[1],"type":"Buffer"},"datatype":4,"dp":1,"fn":0,"seq":12288}'
zigbee2mqtt             | 2021-12-24T18:23:34.089136312Z Zigbee2MQTT:info  2021-12-24 19:23:34: MQTT publish: topic 'zigbee2mqtt/Smoke-Woonkamer', payload '{"battery_low":null,"last_seen":"2021-12-24T19:23:34+01:00","linkquality":255,"smoke":null}'

Generating real smoke alarm with piece of smoking paper (and clearing it again by waving it in fresh air):
Echte smoke alarm:
zigbee2mqtt             | 
zigbee2mqtt             | 2021-12-24T18:48:40.621187380Z Zigbee2MQTT:debug 2021-12-24 19:48:40: Received Zigbee message from 'Smoke-Woonkamer', type 'commandDataReport', cluster 'manuSpecificTuya', data '{"data":{"data":[0],"type":"Buffer"},"datatype":4,"dp":1,"fn":0,"seq":12544}' from endpoint 1 with groupID null
zigbee2mqtt             | 2021-12-24T18:48:40.631655864Z Zigbee2MQTT:debug 2021-12-24 19:48:40: No converter available for 'TS0601_smoke' with cluster 'manuSpecificTuya' and type 'commandDataReport' and data '{"data":{"data":[0],"type":"Buffer"},"datatype":4,"dp":1,"fn":0,"seq":12544}'
zigbee2mqtt             | 2021-12-24T18:48:40.645097338Z Zigbee2MQTT:info  2021-12-24 19:48:40: MQTT publish: topic 'zigbee2mqtt/Smoke-Woonkamer', payload '{"battery_low":null,"last_seen":"2021-12-24T19:48:40+01:00","linkquality":255,"smoke":null}'
zigbee2mqtt             | 2021-12-24T18:48:46.141742694Z Zigbee2MQTT:debug 2021-12-24 19:48:46: Received Zigbee message from 'Smoke-Woonkamer', type 'commandDataReport', cluster 'manuSpecificTuya', data '{"data":{"data":[1],"type":"Buffer"},"datatype":4,"dp":1,"fn":0,"seq":12800}' from endpoint 1 with groupID null
zigbee2mqtt             | 2021-12-24T18:48:46.149343460Z Zigbee2MQTT:debug 2021-12-24 19:48:46: No converter available for 'TS0601_smoke' with cluster 'manuSpecificTuya' and type 'commandDataReport' and data '{"data":{"data":[1],"type":"Buffer"},"datatype":4,"dp":1,"fn":0,"seq":12800}'
zigbee2mqtt             | 2021-12-24T18:48:46.158930820Z Zigbee2MQTT:info  2021-12-24 19:48:46: MQTT publish: topic 'zigbee2mqtt/Smoke-Woonkamer', payload '{"battery_low":null,"last_seen":"2021-12-24T19:48:46+01:00","linkquality":239,"smoke":null}'

By the way: according to tuya site it should also be possible to silence the siren when there is a smoke alarm. We still need a piece of the puzzle for complete definition of this device. I dont have a hub, so unfortunately I cant sniff the protocol.

Do I need to arm the alarm or something? The beeper inside is working perfectly, but I never receive something I can use in my home automation. Have paired them several times, but always same result. Am I missing something?

**Comment by psy0rz** [comment]
I have 3 of these sensors. When paired they DONT give any alarm. Also they dont give any zigbee2mqtt message. (only link quality index)

When i unpair them again they sound the alarm again. Very unreliable, as this should never be possible. Its should always sound the alarm, even if there would be something wrong with zigbee.

Can anyone confirm?

**Comment by ikohlernz** [comment]
Similar behaviour here, the zigbee interview finishes fine, all indication is that device is configured properly, however only link quality shows value, pressing the test button does not indicates smoke. The device seems to other work well.
Here is the debug info:

info Zigbee: allowing new devices to join.
info MQTT publish: topic 'zigbee2mqtt/bridge/response/permit_join', payload '{"data":{"time":254,"value":true},"status":"ok","transaction":"ysdlz-11"}'
info Device '0xa4c1381fbde4d0d8' joined
info MQTT publish: topic 'zigbee2mqtt/bridge/event', payload '{"data":{"friendly_name":"0xa4c1381fbde4d0d8","ieee_address":"0xa4c1381fbde4d0d8"},"type":"device_joined"}'
info MQTT publish: topic 'zigbee2mqtt/bridge/log', payload '{"message":{"friendly_name":"0xa4c1381fbde4d0d8"},"type":"device_connected"}'
info Starting interview of '0xa4c1381fbde4d0d8'
info MQTT publish: topic 'zigbee2mqtt/bridge/event', payload '{"data":{"friendly_name":"0xa4c1381fbde4d0d8","ieee_address":"0xa4c1381fbde4d0d8","status":"started"},"type":"device_interview"}'
info MQTT publish: topic 'zigbee2mqtt/bridge/log', payload '{"message":"interview_started","meta":{"friendly_name":"0xa4c1381fbde4d0d8"},"type":"pairing"}'
debug Device '0xa4c1381fbde4d0d8' announced itself
info MQTT publish: topic 'zigbee2mqtt/bridge/event', payload '{"data":{"friendly_name":"0xa4c1381fbde4d0d8","ieee_address":"0xa4c1381fbde4d0d8"},"type":"device_announce"}'
info MQTT publish: topic 'zigbee2mqtt/bridge/log', payload '{"message":"announce","meta":{"friendly_name":"0xa4c1381fbde4d0d8"},"type":"device_announced"}'
debug Received Zigbee message from '0xa4c1381fbde4d0d8', type 'readResponse', cluster 'genBasic', data '{"modelId":"TS0601"}' from endpoint 1 with groupID 0
debug Skipping message, definition is undefined and still interviewing
debug Received Zigbee message from '0xa4c1381fbde4d0d8', type 'readResponse', cluster 'genBasic', data '{"manufacturerName":"_TZE200_uebojraa"}' from endpoint 1 with groupID 0
debug Skipping message, definition is undefined and still interviewing
debug Received Zigbee message from '0xa4c1381fbde4d0d8', type 'readResponse', cluster 'genBasic', data '{"powerSource":3}' from endpoint 1 with groupID 0
debug Skipping message, definition is undefined and still interviewing
debug Received Zigbee message from '0xa4c1381fbde4d0d8', type 'readResponse', cluster 'genBasic', data '{"zclVersion":3}' from endpoint 1 with groupID 0
debug Skipping message, definition is undefined and still interviewing
debug Received Zigbee message from '0xa4c1381fbde4d0d8', type 'readResponse', cluster 'genBasic', data '{"appVersion":67}' from endpoint 1 with groupID 0
debug Skipping message, definition is undefined and still interviewing
debug Received Zigbee message from '0xa4c1381fbde4d0d8', type 'readResponse', cluster 'genBasic', data '{"stackVersion":0}' from endpoint 1 with groupID 0
debug Skipping message, definition is undefined and still interviewing
debug Received Zigbee message from '0xa4c1381fbde4d0d8', type 'readResponse', cluster 'genBasic', data '{"hwVersion":1}' from endpoint 1 with groupID 0
debug Skipping message, definition is undefined and still interviewing
debug Received Zigbee message from '0xa4c1381fbde4d0d8', type 'readResponse', cluster 'genBasic', data '{"dateCode":""}' from endpoint 1 with groupID 0
debug Skipping message, definition is undefined and still interviewing
debug Received Zigbee message from '0xa4c1381fbde4d0d8', type 'readResponse', cluster 'genBasic', data '{}' from endpoint 1 with groupID 0
debug Skipping message, definition is undefined and still interviewing
info Successfully interviewed '0xa4c1381fbde4d0d8', device has successfully been paired
info Device '0xa4c1381fbde4d0d8' is supported, identified as: TuYa Smoke sensor (TS0601_smoke)
info MQTT publish: topic 'zigbee2mqtt/bridge/event', payload '{"data":{"definition":{"description":"Smoke sensor","exposes":[{"access":1,"description":"Indicates whether the device detected smoke","name":"smoke","property":"smoke","type":"binary","value_off":false,"value_on":true},{"access":1,"description":"Indicates if the battery of this device is almost empty","name":"battery_low","property":"battery_low","type":"binary","value_off":false,"value_on":true},{"access":1,"description":"Link quality (signal strength)","name":"linkquality","property":"linkquality","type":"numeric","unit":"lqi","value_max":255,"value_min":0}],"model":"TS0601_smoke","options":[],"supports_ota":false,"vendor":"TuYa"},"friendly_name":"0xa4c1381fbde4d0d8","ieee_address":"0xa4c1381fbde4d0d8","status":"successful","supported":true},"type":"device_interview"}'
info Configuring '0xa4c1381fbde4d0d8'
info MQTT publish: topic 'zigbee2mqtt/bridge/log', payload '{"message":"interview_successful","meta":{"description":"Smoke sensor","friendly_name":"0xa4c1381fbde4d0d8","model":"TS0601_smoke","supported":true,"vendor":"TuYa"},"type":"pairing"}'
info MQTT publish: topic 'homeassistant/binary_sensor/0xa4c1381fbde4d0d8/smoke/config', payload '{"availability":[{"topic":"zigbee2mqtt/bridge/state"}],"device":{"identifiers":["zigbee2mqtt_0xa4c1381fbde4d0d8"],"manufacturer":"TuYa","model":"Smoke sensor (TS0601_smoke)","name":"0xa4c1381fbde4d0d8"},"device_class":"smoke","json_attributes_topic":"zigbee2mqtt/0xa4c1381fbde4d0d8","name":"0xa4c1381fbde4d0d8 smoke","payload_off":false,"payload_on":true,"state_topic":"zigbee2mqtt/0xa4c1381fbde4d0d8","unique_id":"0xa4c1381fbde4d0d8_smoke_zigbee2mqtt","value_template":"{{ value_json.smoke }}"}'
info MQTT publish: topic 'homeassistant/binary_sensor/0xa4c1381fbde4d0d8/battery_low/config', payload '{"availability":[{"topic":"zigbee2mqtt/bridge/state"}],"device":{"identifiers":["zigbee2mqtt_0xa4c1381fbde4d0d8"],"manufacturer":"TuYa","model":"Smoke sensor (TS0601_smoke)","name":"0xa4c1381fbde4d0d8"},"device_class":"battery","entity_category":"diagnostic","json_attributes_topic":"zigbee2mqtt/0xa4c1381fbde4d0d8","name":"0xa4c1381fbde4d0d8 battery low","payload_off":false,"payload_on":true,"state_topic":"zigbee2mqtt/0xa4c1381fbde4d0d8","unique_id":"0xa4c1381fbde4d0d8_battery_low_zigbee2mqtt","value_template":"{{ value_json.battery_low }}"}'
info MQTT publish: topic 'homeassistant/sensor/0xa4c1381fbde4d0d8/linkquality/config', payload '{"availability":[{"topic":"zigbee2mqtt/bridge/state"}],"device":{"identifiers":["zigbee2mqtt_0xa4c1381fbde4d0d8"],"manufacturer":"TuYa","model":"Smoke sensor (TS0601_smoke)","name":"0xa4c1381fbde4d0d8"},"enabled_by_default":false,"entity_category":"diagnostic","icon":"mdi:signal","json_attributes_topic":"zigbee2mqtt/0xa4c1381fbde4d0d8","name":"0xa4c1381fbde4d0d8 linkquality","state_class":"measurement","state_topic":"zigbee2mqtt/0xa4c1381fbde4d0d8","unique_id":"0xa4c1381fbde4d0d8_linkquality_zigbee2mqtt","unit_of_measurement":"lqi","value_template":"{{ value_json.linkquality }}"}'
debug Received MQTT message on 'homeassistant/binary_sensor/0xa4c1381fbde4d0d8/smoke/config' with data '{"availability":[{"topic":"zigbee2mqtt/bridge/state"}],"device":{"identifiers":["zigbee2mqtt_0xa4c1381fbde4d0d8"],"manufacturer":"TuYa","model":"Smoke sensor (TS0601_smoke)","name":"0xa4c1381fbde4d0d8"},"device_class":"smoke","json_attributes_topic":"zigbee2mqtt/0xa4c1381fbde4d0d8","name":"0xa4c1381fbde4d0d8 smoke","payload_off":false,"payload_on":true,"state_topic":"zigbee2mqtt/0xa4c1381fbde4d0d8","unique_id":"0xa4c1381fbde4d0d8_smoke_zigbee2mqtt","value_template":"{{ value_json.smoke }}"}'
debug Received MQTT message on 'homeassistant/binary_sensor/0xa4c1381fbde4d0d8/battery_low/config' with data '{"availability":[{"topic":"zigbee2mqtt/bridge/state"}],"device":{"identifiers":["zigbee2mqtt_0xa4c1381fbde4d0d8"],"manufacturer":"TuYa","model":"Smoke sensor (TS0601_smoke)","name":"0xa4c1381fbde4d0d8"},"device_class":"battery","entity_category":"diagnostic","json_attributes_topic":"zigbee2mqtt/0xa4c1381fbde4d0d8","name":"0xa4c1381fbde4d0d8 battery low","payload_off":false,"payload_on":true,"state_topic":"zigbee2mqtt/0xa4c1381fbde4d0d8","unique_id":"0xa4c1381fbde4d0d8_battery_low_zigbee2mqtt","value_template":"{{ value_json.battery_low }}"}'
debug Received MQTT message on 'homeassistant/sensor/0xa4c1381fbde4d0d8/linkquality/config' with data '{"availability":[{"topic":"zigbee2mqtt/bridge/state"}],"device":{"identifiers":["zigbee2mqtt_0xa4c1381fbde4d0d8"],"manufacturer":"TuYa","model":"Smoke sensor (TS0601_smoke)","name":"0xa4c1381fbde4d0d8"},"enabled_by_default":false,"entity_category":"diagnostic","icon":"mdi:signal","json_attributes_topic":"zigbee2mqtt/0xa4c1381fbde4d0d8","name":"0xa4c1381fbde4d0d8 linkquality","state_class":"measurement","state_topic":"zigbee2mqtt/0xa4c1381fbde4d0d8","unique_id":"0xa4c1381fbde4d0d8_linkquality_zigbee2mqtt","unit_of_measurement":"lqi","value_template":"{{ value_json.linkquality }}"}'
info Successfully configured '0xa4c1381fbde4d0d8'

**Comment by psy0rz** [comment]
Let me be clear: I tested with actual smoke, and the build in alarm-siren did NOT make a sound. (tested with 3 sensors)

Can you test that as well, since this is actually extremely dangerous: 

# The test button makes the alarm sound, letting you think the thing is fine, while its actually not functioning at all!

(Resetting the pairing fixes them again, and they react to smoke with sound.)

I also tested the Heiman zigbee smoke sensors and they work perfectly. A few euros more expensive, but they feel and look a bit more solid, and they work perfectly with zigbee. Also they are a LOT louder. (annoyingly loud) So i will buy 8 of those for my whole house and other buildings.

**Comment by Roelski** [comment]
With the commit from https://github.com/Koenkk/zigbee-herdsman-converters/pull/3615#issuecomment-1005058757 it is now working for me, although I'm still on dev. Not sure if it is also in main repo.
I also have my tuya hub now and registered an account. I want to find out if the internal buzzer can be switched off as is described on their site. But test button and fire alarm are working correctly with the mentioned commit.
Internal siren of device is always working for me.


// edit
I just connected 2 of my 3 smoke sensors on the tuya hub.
Testbutton: both work and give audible alarm, and in app a smoke warning
Real smoke: 1 of the 2 sensors does NOT make an alarm?!?! Not even on the red button. WTH?!
The one that gave alarm was not able to switch off the audible alarm from the app, although there is a switch for it (maybe for other smoke sensor it will be working).

//Edit 2: opened it up and there was a solder bridge missing on the print. Wow.... so the test button works without this connection, but the actual smoke alarm does not?!? So you know: always test it once with real smoke!!! This one is repaired now.

**Reply by psy0rz** [reply]
Thats the thing that bothers me: if pairing it has somehow influence on its ability to function, it seems very unreliable firmware?

Or is there an actual official way to disable the sound via zigbee? And is our zigbee2mqtt implementation accidentally triggering it?

**Reply by Roelski** [reply]
I see this as two separate things. 
Test button in my case is always working: audible alarm and red buttion lighting. Only reals smoke didnt work on 1 sensor. In that case no alarm anywhere. I at least expected alarm in app.

In your case it looks like the audible alarm is switched off. Is the red button lighting when doing the test?

The smart-life app does have a slider to switch off the siren. But every time I slide it and return to the menu, it is activated again.
In Tyua dev platform I do see the following:

Standard Instruction Set
Code   Type   Values
muffling	Boolean "{true,false}"

Standard Status Set
Code  Type  Values

smoke_sensor_status	Enum	
{
  "range": [
    "alarm",
    "normal"
  ]
}
battery_state	Enum	
{
  "range": [
    "low",
    "middle",
    "high"
  ]
}
muffling	Boolean	"{true,false}"

So It looks like it CAN be set. Then again, I still would expect only the internal buzzer to be silenced, and the light and mqtt reporting still work.

Device logs (what is described to get the id numbers to get new devices to work) dont show me sliding the 'muffling' slider, so i cant see if this can be set with mqtt.
I see I can set the muffling using the Tuya dev portal. I'll try that now.

//edit
Not working for me, although site says it is silenced.

I do see this in logging:
[
    {
        "dpName": "Silence",
        "dpId": 16
    },
    {
        "dpName": "Smoke Detection State",
        "dpId": 1
    },
    {
        "dpName": "tamper_alarm",
        "dpId": 4
    },
    {
        "dpName": "Battery level state",
        "dpId": 14
    }
]

Maybe devs can do something with these id's?


Few test further. Looks like my silence is alway reset, but it is visible in log.
Request pauloas (filtered: Silence)
{
    "startRowId": "",
    "pageNo": 1,
    "pageSize": 10,
    "code": "16",
    "startTime": 1641988316290,
    "endTime": 1642161116291,
    "projectCode": "p1641732862537ghrupc",
    "sourceId": "eu1641730637721XnyVv",
    "sourceType": "4",
    "deviceId": "bf1106eb1314cd77a1hgjt",
    "pageStartRow": "",
    "region": "EU"
}


Preview result:

{
    "devId": "bf1106eb1314cd77a1hgjt",
    "datas": [
        {
            "devId": "bf1106eb1314cd77a1hgjt",
            "eventType": "Publish",
            "sourceDetail": "",
            "eventTime": 1642161080606,
            "eventName": "Silence",
            "eventTimeStr": "2022-01-14 11:51:20:606",
            "eventDetail": "OFF",
            "requestFrom": "unknown"
        },
        {
            "devId": "bf1106eb1314cd77a1hgjt",
            "eventType": "Publish",
            "sourceDetail": "",
            "eventTime": 1642161061948,
            "eventName": "Silence",
            "eventTimeStr": "2022-01-14 11:51:01:948",
            "eventDetail": "Open Mute",
            "requestFrom": "unknown"
        },
        {
            "devId": "bf1106eb1314cd77a1hgjt",
            "eventType": "Publish",
            "sourceDetail": "",
            "eventTime": 1642161056962,
            "eventName": "Silence",
            "eventTimeStr": "2022-01-14 11:50:56:962",
            "eventDetail": "Open Mute",
            "requestFrom": "unknown"
        },
        {
            "devId": "bf1106eb1314cd77a1hgjt",
            "eventType": "Publish",
            "sourceDetail": "",
            "eventTime": 1642161018210,
            "eventName": "Silence",
            "eventTimeStr": "2022-01-14 11:50:18:210",
            "eventDetail": "Open Mute",
            "requestFrom": "unknown"
        },
        {
            "devId": "bf1106eb1314cd77a1hgjt",
            "eventType": "Publish",
            "sourceDetail": "",
            "eventTime": 1642160592041,
            "eventName": "Silence",
            "eventTimeStr": "2022-01-14 11:43:12:041",
            "eventDetail": "Open Mute",
            "requestFrom": "unknown"
        }
    ],
    "docType": 1,
    "currentPageStartRowKey": "YzlmNTFiODV0amdoMWE3N2RjNDEzMWJlNjAxMWZiXzky<partialy removed>"
}

**Comment by Zagitta** [comment]
Can confirm this is still an issue.
It also doesn't seem to report dp 15 with battery % but only battery_state in dp 14.

**Comment by psy0rz** [comment]
I removed the batteries since 2 of the 3 recently gave false alarms. Could it be the sensors got somewhat dirty during all my tests with actual smoke?

Do they sort of recalibrate when you powercycle them perhaps?

Not going to test since one scared the hell out of me at 3am. im done with them.

**Comment by Roelski** [comment]
No problems with sensor in itself. I checked in tuya develop.
Parameters I can see are:
Standard status set:

smoke_sensor_status	Enum	
{
  "range": [
    "alarm",
    "normal"
  ]
}

temper_alarm	Boolean	
"{true,false}"

battery_state	Enum	
{
  "range": [
    "low",
    "middle",
    "high"
  ]
}

muffling	Boolean	
"{true,false}"

The  Muffling parameter I havent been able to get that to work. Shows as Silence pulldown in log.

If I set it I get device event: Publish, DP ID: Silence, Event Details: Open Mute
On device log page: 
Battery state was empty on debug page, but it showed as code 14.
Silence was shown as code 16.

Not sure how to debug set events. Maybe this information can help anyone, although I think this information was already known.
